java/client/src/org/openqa/selenium/net/PortProber.java

```diff
@@ -156,4 +157,22 @@ public static boolean pollPort(int port, int timeout, TimeUnit unit) {
 
     return false;
   }
+
+  public static void waitForPortUp(int port, int timeout, TimeUnit unit) {
+    long end = System.currentTimeMillis() + unit.toMillis(timeout);
+    while (System.currentTimeMillis() < end) {
+      try {
+        Socket socket = new Socket();
+        socket.connect(new InetSocketAddress("localhost", port), 1000);
```

Comment:
so, why not modify the other pollPort? I'm not a fan of seemingly duplicate code that does the same thing